pkg/pipeline/run.go

```diff
@@ -0,0 +1,116 @@
+/*
+ Copyright 2021 The Crossplane Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package pipeline
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/crossplane-contrib/terrajet/pkg/config"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+)
+
+// Run runs the Terrajet code generation pipelines.
+func Run(pc *config.Provider) { // nolint:gocyclo
```

Comment:
Would it make sense to call `pc.ConfigureResources()` here in this function?

Reply:
Initially, it was like that, but, we need to do some at provider runtime since we are using config there as well. So, moved it to a common place which is `config/provider.go` which I am also not that happy with.